spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/container/ContainerImage.java

```diff
@@ -155,19 +162,52 @@ public void setRepositoryName(String repositoryName) {
 		this.repositoryName = repositoryName;
 	}
 
+	public String getRepositoryReference() {
+		return (StringUtils.hasText(this.repositoryTag) ? repositoryTag : repositoryDigest);
+	}
+
+	private String getReferencePrefix() {
+		if (getRepositoryReferenceType() == RepositoryReferenceType.digest) {
+			return "@";
+		}
+		return ":";
+	}
+
+	public RepositoryReferenceType getRepositoryReferenceType() {
+		if (StringUtils.hasText(this.repositoryTag)) {
+			return RepositoryReferenceType.tag;
+		} if (StringUtils.hasText(this.repositoryDigest)) {
+			return RepositoryReferenceType.digest;
+		}
+		return RepositoryReferenceType.unknown;
+	}
+
 	public String getRepositoryTag() {
 		return repositoryTag;
 	}
 
 	public void setRepositoryTag(String repositoryTag) {
-		Assert.isTrue(TAG_PATTERN.matcher(repositoryTag).matches(),
-				"Invalid repository tag: " + repositoryTag);
+		Assert.isTrue(TAG_PATTERN.matcher(repositoryTag).matches(), "Invalid repository tag: " + repositoryTag);
+		Assert.isTrue(!StringUtils.hasText(this.repositoryDigest),
+				"Can set repository Tag because of existing Digest " + repositoryDigest);
```

Comment:
you mean `Can't set repository Tag` right?

Reply:
It is `Can't` indeed